Added fill color for progress

wassafshahzad opened the pull request:
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired changed?

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update


### New Feature Submission:

- [ ] Does your submission pass the tests? 
- [ ] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

### Description
Added a fill_color  property to update the background style of the fill in progress bar. Since we do not expose the Progress.Root and Progress.Indicator components of radix ui. I updated the indicator class directly.

Lendemor reviewed (dismissed):
Tested, it's working well.

Just need to fix the pre-commit issue before we can merge it.

wassafshahzad commented:
> Tested, it's working well.
> 
> Just need to fix the pre-commit issue before we can merge it.

I unfortunetly dont have the .pyi_generator_last_run and thus generating the pyi will cause issue for me

Lendemor commented:
> I unfortunetly dont have the .pyi_generator_last_run and thus generating the pyi will cause issue for me

I tested locally to delete `.pyi_generator_last_run` and running `py scripts/make_pyi.py reflex/components/radix/themes/components/progress.py` worked without issue.

wassafshahzad commented:
> python3 scripts/make_pyi.py reflex/components/radix/themes/components/progress.py

Done. Thank you

wassafshahzad commented:
> > I unfortunetly dont have the .pyi_generator_last_run and thus generating the pyi will cause issue for me
> 
> I tested locally to delete `.pyi_generator_last_run` and running `python3 scripts/make_pyi.py reflex/components/radix/themes/components/progress.py` worked without issue.

@Lendemor  Addressed feedback

Lendemor commented:
Approved, thanks for the good work !
